### PR TITLE
Add theme to ButtonGroup

### PR DESCRIPTION
**Pull request by jtanner**
_Targets: `joe/theme-master` branch for #588_

* Adds theme to ButtonGroup
* Fixes bug with `adjustColor` where it didn't handle 3 character hex values

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -58,7 +56,8 @@ class Button extends React.Component {
   };
 
   render () {
-    const styles = this.styles();
+    const theme = StyleUtils.mergeTheme(this.props.theme, this.props.primaryColor);
```
Comment (lukeschunk) [inline src/components/Button.js]: eventually i think we could kill `this.props.primaryColor` and just use the theme

**Comment by lukeschunk** [comment]
:shipit:

File: src/components/Button.js (hunk 1 above)
Comment (jtanner) [inline src/components/Button.js]: Yes, that's the idea 😄 

See https://github.com/mxenabled/mx-react-components/pull/594/files#diff-734f11edfaa69e2a96b8cb8f7f06fb6bR34

And https://github.com/mxenabled/mx-react-components/pull/594/files#diff-ec85bd453c96420d8df62f4d96bd694aR33

Comment (lukeschunk) [inline src/components/Button.js]: yeah i realized after reviewiing a few more of these that you were way ahead of me 🙃

**Pull request merged**